features/graph/api/src/test/java/org/opennms/netmgt/graph/api/generic/GenericEdgeTest.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.graph.api.generic;
+
+import static org.junit.Assert.*;
+
+import java.util.HashMap;
+import java.util.function.Supplier;
+
+import org.junit.Test;
+
+public class GenericEdgeTest {
```

Comment:
We should also verify, that when adding an edge to a graph either the source or target matches the graph's namespace.

Reply:
+1
done here: https://github.com/OpenNMS/opennms/blob/9e4067ebc10a759582aa2337fe56a91dc5ee43dc/features/graph/api/src/main/java/org/opennms/netmgt/graph/api/generic/GenericGraph.java#L200
and tested here:
https://github.com/OpenNMS/opennms/blob/9e4067ebc10a759582aa2337fe56a91dc5ee43dc/features/graph/api/src/test/java/org/opennms/netmgt/graph/api/generic/GenericGraphTest.java#L47